symengine/dict.h

```diff
@@ -185,6 +185,165 @@ std::ostream &operator<<(std::ostream &out, const SymEngine::vec_basic &d);
 std::ostream &operator<<(std::ostream &out, const SymEngine::set_basic &d);
 std::ostream &operator<<(std::ostream &out, const SymEngine::map_int_Expr &d);
 
+// dict wrapper
+template <typename Key, typename Value, typename Wrapper>
+class ODictWrapper
```

Comment:
Why? This is a general dictionary wrapper, which maybe used elsewhere, right? Specially because the `dict.h` file contains other templates and typedefs.